util/flightcontrol/flightcontrol.go

```diff
@@ -262,7 +262,9 @@ func (sc *sharedContext[T]) checkDone() bool {
 	for _, ctx := range sc.ctxs {
 		select {
 		case <-ctx.Done():
-			err = ctx.Err()
+			// Cause can't be used here because this error is returned for Err() in custom context
+			// implementation and unfortunately stdlib does not allow defining Cause() for custom contexts
+			err = ctx.Err() //nolint: forbidigo
```

Comment:
Looks like there is no way to define `Cause()` atm for custom implementations of context interface so this needs to return `Err()` directly. Maybe this can be improved in the future if we could avoid defining `Err()` and instead would use embedded stdlib context. But I think we can look at this case separately.